### PR TITLE
Added the option to define aliases for the documentation by the user.

### PR DESCRIPTION
**Pull request by haraldkl**
This enables the extension of the dictionary used in sub_macros by definitions of aliases from the user.
The definition in the project takes the following form:
alias: key = text/to/replace
As with the predefined macro definitions, this then can be used in the documentation with |key|.
Eliminates the base_url argument from the sub_macros routine.

### CODE REVIEW

File: ford/__init__.py [hunk 1]
```diff
@@ -355,15 +356,30 @@ def main(proj_data,proj_docs,md):
...
+                                    os.path.join(proj_data['project_url'],
+                                                 'media') ) )
+    ford.utils.register_macro( 'page = {0}'.format(
+                                    os.path.join(proj_data['project_url'],
+                                                 'page') ) )
+    # Register the user defined aliases:
+    for alias in proj_data['alias']:
+        val, key = ford.utils.register_macro(alias)
+        if key == None:
+            raise RuntimeError('Error, no alias name provided for {0}'.format(val))
```
Comment (ZedThree) [inline ford/__init__.py:372]: Might be better to just move this into `register_macro`?

File: ford/__init__.py [hunk 2]
```diff
@@ -355,15 +356,30 @@ def main(proj_data,proj_docs,md):
...
+    ford.utils.register_macro('url = {0}'.format(proj_data['project_url']))
+    ford.utils.register_macro( 'media = {0}'.format(
+                                    os.path.join(proj_data['project_url'],
+                                                 'media') ) )
+    ford.utils.register_macro( 'page = {0}'.format(
+                                    os.path.join(proj_data['project_url'],
+                                                 'page') ) )
+    # Register the user defined aliases:
+    for alias in proj_data['alias']:
+        val, key = ford.utils.register_macro(alias)
```
Comment (ZedThree) [inline ford/__init__.py:370]: Are these used outside of the check on the next line? Could just drop them entirely?

**Review by ZedThree — COMMENTED** [review]
Nice, thanks @haraldkl ! Just a couple of comments -- I'm happy to fix these myself if you like

File: ford/__init__.py (hunk 1 above)
Comment (haraldkl) [reply ford/__init__.py:372]: Sure, please make changes as you see fit.

File: ford/__init__.py (hunk 2 above)
Comment (haraldkl) [reply ford/__init__.py:370]: Sorry, I fear I don't get the question. What these are you refering to? The alias loop?

Comment (ZedThree) [reply ford/__init__.py:370]: Sorry, I meant `val, key` -- I just completely didn't type them, and somehow expected you to know what I was thinking!

Comment (haraldkl) [reply ford/__init__.py:370]: Yes, sorry. Didn't connect the dots. Yes, moving the check into the subroutine should render them useless here. Only the `_MACRO_DICT` should be used elsewhere.

**Review by ZedThree — APPROVED** [review]

**Comment by haraldkl** [comment]
Thanks a lot! In #197 I mentioned that we are using references to external documentations, to split a larger project into subprojects to reduce the project sizes of individual projects. I maintain this as a branch: https://github.com/haraldkl/ford/tree/external. Do you think it would be of interest to create a pull request for this?

**Comment by ZedThree** [comment]
Yep, sounds interesting!

**Pull request merged**